lib/streamlit/Report.py

```diff
@@ -64,13 +64,23 @@ def __init__(self, script_path, argv):
         argv : list of str
             Command-line arguments to run the script with.
 
+        command_line : string
+            Command line as input by the user
+
         """
-        basename = os.path.basename(script_path)
 
         self.script_path = os.path.abspath(script_path)
         self.script_folder = os.path.dirname(self.script_path)
         self.argv = argv
-        self.name = os.path.splitext(basename)[0]
+        self.command_line = command_line
+
+        if len(command_line.split()) > 2:
+            # Extract script name in a command line of the form
+            # streamlit run script.py
+            basename = os.path.basename(command_line.split()[2])
+            self.name = os.path.splitext(basename)[0]
+        else:
+            self.name = "script"
```

Comment:
In what situations will `name` be `script`?

This seems like a weird name, we should come up with a better one. Maybe just "Streamlit app"?

Reply:
I removed this